File: ios/Plugin/CryptoPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@available(iOS 13.0, *)
@objc(CryptoPlugin)
public class CryptoPlugin: CAPPlugin {
    private let implementation = Crypto()

    @objc func encrypt(_ call: CAPPluginCall) {
        guard let base64Encoded = call.getString("base64Encoded") else {
            call.reject("'base64Encoded' value must be provided")
            return
        }
        
        guard let text = call.getString("text") else {
            call.reject("'text' value must be provided")
            return
        }
        
        implementation.encrypt(text: text, base64Encoded: base64Encoded) { (result, success) in
            
            if success {
                call.resolve([
                    "encrypted": result
                ])
            } else {
                call.reject(result)
            }
        }
    }

    @objc func decrypt(_ call: CAPPluginCall) {
        guard let base64Encoded = call.getString("base64Encoded") else {
            call.reject("'base64Encoded' value must be provided")
            return
        }
        
        guard let text = call.getString("text") else {
            call.reject("'text' value must be provided")
            return
        }
        
        implementation.decrypt(text: text, base64Encoded: base64Encoded) { (result, success) in
            if success {
                call.resolve([
                    "decrypted": result
                ])
            } else {
                call.reject(result)
            }
        }
    }
    
    @objc func generateSymmetricKey(_ call: CAPPluginCall) {
        call.resolve([
            "key256": implementation.generateSymmetricKey()
        ])
    }
}

File: android/src/main/java/com/cynx/crypto/CryptoPlugin.java
package com.cynx.crypto;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Crypto")
public class CryptoPlugin extends Plugin {

    private Crypto implementation = new Crypto();

    @PluginMethod
    public void encrypt(PluginCall call)  {
        String base64Encoded = call.getString("base64Encoded");
        String text = call.getString("text");
        String iv = call.getString("encodedIV");

        if (base64Encoded == null || text == null || iv == null) {
            call.reject("encrypt:  params missing");
        }

        JSObject ret = new JSObject();
        try {
            ret.put("value", implementation.encrypt(text, base64Encoded, iv));
            call.resolve(ret);
        } catch (Exception e) {
            call.reject(e.getLocalizedMessage());
        }
    }

    @PluginMethod
    public void decrypt(PluginCall call) {
        String base64Encoded = call.getString("base64Encoded");
        String text = call.getString("text");
        String iv = call.getString("encodedIV");

        if (base64Encoded == null || text == null || iv == null) {
            call.reject("encrypt:  params missing");
        }

        JSObject ret = new JSObject();
        try {
            ret.put("value", implementation.decrypt(text, base64Encoded, iv));
            call.resolve(ret);
        } catch (Exception e) {
            call.reject(e.getLocalizedMessage());
        }
    }

    @PluginMethod
    public void generateSymmetricKey(PluginCall call) {
        try {
            JSObject ret = new JSObject();
            ret.put("value", implementation.generateSymmetricKey());
            call.resolve(ret);
        } catch (Exception e) {
            call.reject(e.getLocalizedMessage());
        }
    }

    @PluginMethod
    public void generateIV(PluginCall call) {
        try {
            JSObject ret = new JSObject();
            ret.put("value", implementation.generateIV());
            call.resolve(ret);
        } catch (Exception e) {
            call.reject(e.getLocalizedMessage());
        }
    }
}
